src/Core/Services/MetadataProviders/ISqlMetadataProvider.cs

```diff
@@ -53,6 +53,11 @@ public interface ISqlMetadataProvider
         /// </summary>
         FieldDefinitionNode? GetSchemaGraphQLFieldFromFieldName(string entityName, string fieldName);
 
+        /// <summary>
+        /// Gets a collection of linking entities generated by DAB (required to support nested mutations).
+        /// </summary>
+        IReadOnlyDictionary<string, Entity> GetLinkingEntities() => new Dictionary<string, Entity>();
+
```

Comment:
I am not very sure if it'll be worth it just now. Without downcasting it to a dictionary, it'll still not be possible to mutate this dictionary. However, do you still suggest we use an immutable dictionary to prevent this case (as a defensive approach)- no matter how unlikely it seems for now?